internal/fetch_helper.go

```diff
@@ -169,6 +169,14 @@ func putCachedDecompressorInFirstPlace(decompressors []compression.Decompressor)
 
 // TODO : unit tests
 func DownloadAndDecompressStorageFile(folder storage.Folder, fileName string) (io.ReadCloser, error) {
+	reader, exists, err := TryDownloadFile(folder, fileName)
```

Comment:
What's the purpose of this?

It seems like no compression should be the last resort